File: set files/set.cpp
#include<bits/stdc++.h>

using namespace std;

int main ()
{
    int n = 10*1000;
    freopen("vector104.txt","r",stdin);
    int v[n];
    for(int i = 0; i < n; i++){
        int num;
        cin>>num;
        v[i] = num;
    }
    fclose(stdin);
    auto start = std::chrono::system_clock::now();
    set<int >s;
    for(int i = 0; i < n; i++){
        s.insert(v[i]);
    }
    auto end = chrono::system_clock::now();
    chrono::duration<double> elapsed_seconds = end-start;
    cout<<"Elapsed time: "<<elapsed_seconds.count()<<"\n";
    return 0;
}

File: set files/set.rb
require 'set'
time = Time.now.to_f
v = Set.new
File.open("vector105.txt", "r") do |f|
    f.each_line do |line|
        v << (line.to_i)
    end
end
time = Time.now.to_f - time
puts time